pac4j-core/src/main/java/org/pac4j/core/client/IndirectClient.java

```diff
@@ -40,7 +40,7 @@ protected void internalInit(final WebContext context) {
     }
 
     @Override
-    public final HttpAction redirect(final WebContext context) throws HttpAction {
+    public HttpAction redirect(final WebContext context) throws HttpAction {
```

Comment:
In your case, I think you can override the `getAccessToken` method and create the appropriate Scribe service based on the tenant callback URL to retrieve the token: https://github.com/pac4j/pac4j/blob/1.9.x/pac4j-oauth/src/main/java/org/pac4j/oauth/client/BaseOAuth20Client.java#L55